### PR TITLE
GitHub test coverage improvements

### PR DESCRIPTION
**Pull request by raymondberg**
## Problem

Test Coverage in the Github Summary work!

## Solution

Refactor the code a bit to push some logic into the lists. Make the necessary information for a list, including formatting and 

busy_beaver/tasks/github_stats/summary.py | 38 | 6 | 0 | 84%
Total | 862 | 84 | 1 | 90%

busy_beaver/tasks/github_stats/summary.py | 22 | 0 | 0 | 100%
Total | 857 | 59 | 2 | 93%

I think there's some more work to be done with the noun formatting. Would like to look into redoing with [`inflect`](https://pypi.org/project/inflect/).

### CODE REVIEW

File: busy_beaver/tasks/github_stats/event_list.py [hunk 1]
```diff
@@ -1,19 +1,17 @@
-import gettext
+import inflect
```
Comment (alysivji) [inline busy_beaver/tasks/github_stats/event_list.py]: Thoughts on making this into an adapter?

File: busy_beaver/tasks/github_stats/event_list.py [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
 class EventList:
+    EMOJI = ":heart:"
+    NOUN = "event"
+
     def __init__(self):
         self.events = []
 
     def __repr__(self):
-        return repr(self.events)
+        return repr(self.events)  # pragma: no cover
```
Comment (alysivji) [inline busy_beaver/tasks/github_stats/event_list.py]: Thanks! I've been meaning to do this forever.

File: busy_beaver/tasks/github_stats/summary.py [hunk 3]
```diff
@@ -17,24 +17,27 @@
...
+            IssuesOpenedList(),
+            CommitsList(),
+            StarredReposList(),
+        ]
 
         timeline = github.user_activity_after(user.github_username, boundary_dt)
-        self._classify_events(timeline)
+        for event in timeline:
+            for event_list in self.event_lists:
+                if event_list.matches_event(event):
```
Comment (alysivji) [inline busy_beaver/tasks/github_stats/summary.py]: This is nicely done! Hide all the complexity in your classes and make the rest of your life readable.

Will be using this a lot.

**Comment by alysivji** [comment]
Thanks @raymondberg!!! This is awesome! Learned a ton about how to use polymorphism to make code even more readable!

I'm going to merge this in, what are your thoughts on making inflect into an adapter? If we decide to go down that route, could you make an issue on the tracker?

**Comment by raymondberg** [comment]
I'm interested, but I'm not sure I know what you mean by an adapter for Inflect.

**Pull request merged**